### PR TITLE
Add counters for downsampling

### PR DESCRIPTION
**Pull request by lkts**
This PR adds counters for downsampling: success, failure and failure due to invalid configuration. They will be used in TSDB dashboard to assess health of downsampling functionality.

### CODE REVIEW

File: x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/DownsampleMetrics.java [hunk 1]
```diff
@@ -31,6 +31,9 @@ public class DownsampleMetrics extends AbstractLifecycleComponent {
 
     public static final String LATENCY_SHARD = "es.tsdb.downsample.latency.shard.histogram";
     public static final String LATENCY_TOTAL = "es.tsdb.downsample.latency.total.histogram";
+    public static final String SUCCESS = "es.tsdb.downsample.success.total";
+    public static final String FAILURE = "es.tsdb.downsample.failure.total";
+    public static final String INVALID_CONFIGURATION = "es.tsdb.downsample.invalid_configuration.total";
```
Comment (kkrik-es) [inline x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/DownsampleMetrics.java:36]: I'd have a sinlge counter "es.tsdb.downsample.actions.total.counter" and "es.tsdb.downsample.actions.shard.counter" and use the dimension to break them down by status. Thoughts?

Comment (martijnvg) [reply x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/DownsampleMetrics.java:36]: I think these new metrics are are total metrics, right? (in other words not per shard)

Comment (kkrik-es) [reply x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/DownsampleMetrics.java:36]: Indeed, I think we need metrics for both total and per-shard actions.

File: x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/DownsampleMetrics.java [hunk 2]
```diff
@@ -43,6 +45,8 @@ protected void doStart() {
         // Register all metrics to track.
         meterRegistry.registerLongHistogram(LATENCY_SHARD, "Downsampling action latency per shard", "ms");
         meterRegistry.registerLongHistogram(LATENCY_TOTAL, "Downsampling latency end-to-end", "ms");
+        meterRegistry.registerLongCounter(ACTIONS_SHARD, "Number of shard-level downsampling actions", "count");
+        meterRegistry.registerLongCounter(ACTIONS, "Number of index-level downsampling actions", "count");
```
Comment (martijnvg) [inline x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/DownsampleMetrics.java:49]: Maybe rename to `Number of index-level downsampling actions` to `Number of downsampling operations`?

**Review by martijnvg — APPROVED** [review]
lgtm

**Review by kkrik-es — APPROVED** [review]

**Pull request merged**